app/models/spree/order.rb

```diff
@@ -590,6 +594,13 @@ def fee_handler
       @fee_handler ||= OrderFeesHandler.new(self)
     end
 
+    def clear_legacy_taxes!
+      # For instances that use additional taxes, old orders can have taxes recorded in
+      # lump-sum amounts per-order. We clear them here before re-applying the order's taxes,
+      # which will now be applied per-item.
+      adjustments.legacy_tax.delete_all
```

Comment:
can you clarify what is "use additional taxes"?
can you clarify what are "old orders"? Orders before a specific version? Orders created before this PR?